packages/core/src/plugins/socketIO/router/adapter.ts

```diff
@@ -1,24 +1,37 @@
 import _debug = require('debug')
-import noop = require('lodash/noop')
-import { ActionTransport } from '../../..'
-import { ServiceRequest } from '../../../types'
+import { ActionTransport, ServiceRequestInterface } from '../../..'
 import { Router } from '../../router/factory'
 import { RequestCallback } from '../../router/dispatcher'
+import { createServiceRequest } from './service-request-factory'
 
 const debug = _debug('mservice:router:socket.io')
 const { socketIO } = ActionTransport
 
+export interface SocketIOMessageOptions {
+  simpleResponse?: boolean;
+}
+
 export interface SocketIOMessage {
-  data: [string, any, RequestCallback];
+  data: [string, any, SocketIOMessageOptions, RequestCallback];
 }
 
-/* Decrease request count on response */
-function wrapCallback(router: Router, callback: RequestCallback) {
+function wrapCallback(
+  router: Router,
+  serviceRequest: ServiceRequestInterface,
+  options: SocketIOMessageOptions,
+  callback: RequestCallback
+) {
   return (err: any, result?: any) => {
+    // Decrease request count on response
     router.requestCountTracker.decrease(socketIO)
-    if (callback) {
-      callback(err, result)
-    }
+
+    if (!callback) return
+
+    const response = options && options.simpleResponse === false
```

Comment:
I suggest to get rid of the `simpleResponse` condition and return an object with scheme:  ` { data, headers }` using approach below:
```
import { isNil } from 'lodash/fp';
const omitNullOrUndefined = omitBy(isNil);
// ...
callback(err, omitNullOrUndefined(response))

```